core/src/test/java/org/elasticsearch/search/aggregations/InternalMultiBucketAggregationTestCase.java

```diff
@@ -57,7 +72,10 @@ public void setUp() throws Exception {
 
     @Override
     protected final T createTestInstance(String name, List<PipelineAggregator> pipelineAggregators, Map<String, Object> metaData) {
-        return createTestInstance(name, pipelineAggregators, metaData, subAggregationsSupplier.get());
+        T instance = createTestInstance(name, pipelineAggregators, metaData, subAggregationsSupplier.get());
+        assert instance.getBuckets().size() <= maxNumberOfBuckets() :
```

Comment:
I thought about something like that too but I didn't manage to implement it because some aggs need to initialize the buckets in the setUp method while the others can just generate them in the createInstance() method. So a common createBucket() method doesn't fit very well and complicates things a bit. I'm going to merge the PR as it is now.